libcaf_test/caf/test/runner.cpp

```diff
@@ -13,23 +13,64 @@
 #include "caf/config_option_adder.hpp"
 #include "caf/config_option_set.hpp"
 #include "caf/detail/log_level.hpp"
+#include "caf/detail/set_thread_name.hpp"
 
+#include <condition_variable>
 #include <iostream>
+#include <mutex>
 #include <optional>
 #include <regex>
 #include <string>
+#include <thread>
 
 namespace caf::test {
 
 namespace {
 
+class watchdog {
+public:
+  watchdog(int secs) {
+    using detail::format_to;
+    thread_ = secs <= 0 ? std::thread() : std::thread{[=] {
```

Comment:
At least to me, the ternary operator makes the control code somewhat hard to follow. There's also a lot going on in the constructor that's not actually related to constructing the object. I think writing it like this makes the intent much more clear:

```suggestion
    if (secs > 0)
      thread_ = std::thread{[this, secs] { run(secs); });
```

And the current body of the lambda should go to a private `run` member function.